### ISSUE TITLE
Editing an entry is applied only after 2 edits

### ISSUE THREAD

**Issue by nottheend** [SecUSo/privacy-friendly-food-tracker#77]
Regurlary I experience the following behaviour:

When I edit an existing entry and save it, the calories will not update according to the new value.
_Expected_ : calories are newly calculated and shown in the daily overview for that particular entry.

_Actual behaviour_ : Only once I edited the same entry the second time, the calories value will be updated in the list view of the day.

Is that a known issue already?

**Comment by Kamuno** [comment]
We'll investigate it. Thanks for reporting.

**Comment by nottheend** [comment]
Cool, thanks!

Also discovered that same goes for deleting: I need to delete an entry twice before it is remove

**Comment by StickyDigit** [comment]
I've been noticing this behaviour too. Today I stumbled on something which may aid diagnosis.

1: Today I made an entry of 125g of red wine (shame.. oh the shame!)

2: I then added 125g of wine to that entry, attempting to edit/update it to 250g (wicked!)
The data did not update.. nothing new there.

3: Then I figured I was sharing the bottle, and might as well make it 375g on the second attempt to update. (no shame left!)
On doing that.. it bumped the data to 250

4: I did the 375 again and it stuck.
Seems like it's buffering stuff oddly/badly.

Sometimes it seems to work ok. Mostly it "appears to take two goes" to add or remove stuff from an existing entry.

Thought I'd best offer this info before that third glass :-)
